trin-storage/src/versioned/mod.rs

```diff
@@ -0,0 +1,33 @@
+pub mod memory;
+pub mod sql;
+pub mod store;
+mod utils;
+
+use rusqlite::types::{FromSql, FromSqlError, ValueRef};
+use strum::{AsRefStr, Display, EnumIter, EnumString};
+
+pub use store::VersionedContentStore;
+pub use utils::create_store;
+
+#[derive(Copy, Clone, Debug, Display, Eq, PartialEq, EnumString, AsRefStr, EnumIter)]
+#[strum(serialize_all = "snake_case")]
+pub enum ContentType {
+    Beacon,
```

Comment:
These are mostly placeholders for now. Their purpose is to define the type of content that is stored in one _store_, and for store to differentiate between types (e.g. different column in a table, or completely different table name).
@ogenev Maybe Beacon should have multiple types?

Reply:
We currently store four types for beacon:  
- `LightCllentBootstrap` (in `content_data` table, but we will need to purge any values older than the subjectivity period (a few months))
- `LightClientUpdate` (in `lc_update`  table)
- `LightClientOptimisticUpdate/LightClientFIialityUpdate` (those are stored  in a cache, because we keep only the latest values).

Reply:
So for the `LightCllentBootstrap`, there is no concept of distance (in deciding whether we store) or storage limit?

Meaning, we should store all fresh content and discard/purge all older content, right?